lib/tower_cli/resources/schedule.py

```diff
@@ -0,0 +1,125 @@
+                if unified_job_template is None:
+                    unified_job_template = (item, jt_id)
+                else:
+                    raise exc.UsageError(
+                        'More than one unified job template fields provided, '
+                        'please tighten your criteria.'
+                    )
+        if unified_job_template is not None:
+            kwargs['unified_job_template'] = unified_job_template[1]
+            return '/'.join([UNIFIED_JT[unified_job_template[0]],
+                             str(unified_job_template[1]), 'schedules/'])
+        elif is_create:
+            raise exc.UsageError('You must provide exactly one unified job'
+                                 ' template field during creation.')
+
+    def decorator_without_pk(obj, *args, **kwargs):
+        old_endpoint = obj.endpoint
+        new_endpoint = helper(kwargs)
+        if is_create:
+            obj.endpoint = new_endpoint
+        result = func(obj, *args, **kwargs)
+        obj.endpoint = old_endpoint
+        return result
+
+    def decorator_with_pk(obj, pk=None, *args, **kwargs):
+        old_endpoint = obj.endpoint
+        new_endpoint = helper(kwargs)
+        if is_create:
+            obj.endpoint = new_endpoint
+        result = func(obj, pk=pk, *args, **kwargs)
+        obj.endpoint = old_endpoint
+        return result
+
+    decorator = decorator_with_pk if has_pk else decorator_without_pk
+    for item in CLICK_ATTRS:
+        setattr(decorator, item, getattr(func, item, []))
+    decorator.__doc__ = func.__doc__
+
+    return decorator
+
+
+class Resource(models.Resource):
+    cli_help = 'Manage schedules within Ansible Tower.'
+    endpoint = '/schedules/'
+
+    # General fields.
+    name = models.Field(unique=True)
+    description = models.Field(required=False, display=False)
+
+    # Unified jt fields. note these fields will only be used during creation.
+    # Plus, one and only one field should be provided.
+    job_template = models.Field(type=types.Related('job_template'),
+                                required=False)
+    inventory_source = models.Field(type=types.Related('inventory_source'),
+                                    required=False)
+    project = models.Field(type=types.Related('project'), required=False)
+
+    # Schedule-specific fields.
+    enabled = models.Field(required=False, type=click.BOOL, default=True,
+                           help_text='Whether this schedule will be used')
+    rrule = models.Field(required=False, display=False,
+                         help_text='Schedule rules specifications which is'
+                         ' less than 255 characters.')
+    extra_data = models.Field(type=models.File('r', lazy=True), required=False,
+                              display=False, help_text='Extra data for '
+                              'schedule rules in the form of a .json file.')
+
+Resource.create = jt_aggregate(Resource.create, is_create=True)
```

Comment:
It seems like it would work to do

```python
Resource.create = resources.command(jt_aggregate(Resource.create, is_create=True))
```

as opposed to putting the `value._cli_command_attrs.setdefault(attkey, attval)` in a try-except block.